Namitha: Enhance/comparing villages chart

Namitha7070 opened the pull request:
# Description
(PRIORITY HIGH) Jae/Jaiwanth: Listing and Bidding Dashboard: Create pie chart titled ‘Comparing Villages’ (WIP Namitha)
<img width="777" height="724" alt="image" src="https://github.com/user-attachments/assets/396da0d2-bce9-4a01-b730-9fe1cda4db83" />

<img width="726" height="552" alt="image" src="https://github.com/user-attachments/assets/7bef5c1b-104b-43cc-a6cd-bfc2a02a9439" />


## Related PRS (if any):
…

## Main changes explained:

- Enhanced ComparePieChart Component: Added comprehensive filter system with date range pickers (defaulting to last 365 days), Compare By dropdown (Villages/Properties), and Type selector (All/Listing/Bidding).
- Visual Layout Redesign: Restructured pie chart to display absolute values ON the pie slices and percentages with village/property names OUTSIDE the chart with connecting label lines, matching the specified design requirements.
- Dynamic Features: Implemented loading states with skeleton UI, error handling with retry functionality, and responsive design optimizations for mobile/tablet devices.
- Dashboard Integration: Connected the enhanced pie chart with the main dashboard's metric selection system, enabling seamless metric switching and proper state management across components.

## How to test:

1. check into current branch
2. do npm install and ... to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to /lbdashboard
6. Check if pie chart is visible and all the requirements

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/bcd6b9f2-6dbc-47d0-96a8-b2d5c7eb8f46

Commit:
add pie chart

Namitha7070 committed:
Implement enhanced ComparePieChart with filters and dynamic data

Prem203 reviewed (changes requested):
Reviewed the PR with an admin account. The graph is rendering correctly in the normal view, but when filters are changed sometimes the chart notations are disappearing partially out of of border. The design is not responsive and the same issue appears in the mobile view.

Changes are required to correctly render the graph in all format. 

<img width="1899" height="907" alt="image" src="https://github.com/user-attachments/assets/2a2c6254-2cea-4856-b4bd-b425850b8b81" />
<img width="1902" height="946" alt="image" src="https://github.com/user-attachments/assets/f9f28827-2f1d-40b8-85f0-4efb6f1125c8" />
<img width="1898" height="960" alt="image" src="https://github.com/user-attachments/assets/01f9ef31-506c-4d78-9a01-d9c2bd4fb41e" />
<img width="1127" height="961" alt="image" src="https://github.com/user-attachments/assets/b4a4ffe1-fe2b-4293-93db-7bf7759af445" />

Anusha-Gali reviewed (changes requested):
Hi Namitha,

I have reviewed your PR locally, and no matter what metric are selected the values in the graph remain same and only change between Villages and Properties in Compare By which i am assuming is due to default data.
The UI issues are:
- All the dates, compare by and type options are cropped out after selection.
- The chart notations are cropped off no matter which option is selected in normal and mobile responsive
- The dark mode is not applied to chart.
Note: i was able to access the graph using volunteer account as well
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 39 48 PM" src="https://github.com/user-attachments/assets/ed1b8a80-3ac9-47de-82b6-66187bcef37d" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 40 00 PM" src="https://github.com/user-attachments/assets/e6db5d32-1397-4d59-99ea-7d5a84d99f7d" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 40 29 PM" src="https://github.com/user-attachments/assets/b59dee5f-786e-4f61-a3d2-6cbf4ae5efac" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 40 42 PM" src="https://github.com/user-attachments/assets/a3d03e40-8dd9-48cb-93ea-88e776a81de3" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 44 35 PM" src="https://github.com/user-attachments/assets/b4b1c377-68a3-4979-9468-20c49c777ed1" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 45 11 PM" src="https://github.com/user-attachments/assets/886df3ca-6d78-4a40-81ae-85d4300c075d" />

rohanrastogi311 reviewed (changes requested):
Hi Namitha, 

Well done on this implementation. A small problem, when changing filters the labels and clipping through borders. 

<img width="1728" height="1039" alt="PR 4436 Screenshot" src="https://github.com/user-attachments/assets/bbc4c5cd-0c5a-4fc9-af41-d85424fb517b" />
<img width="1726" height="1037" alt="PR 4436 Screenshot 2" src="https://github.com/user-attachments/assets/b5fb0fd2-5c10-44f8-9079-8fe22c41286e" />

HemanthNidamanuru reviewed (changes requested):
Hi Namitha,
Thanks for the PR. I pulled it locally and tested it, and here's what I found:

Regardless of which metric is selected, the graph values stay the same. Only the Compare By toggle between Villages and Properties changes anything, which I'm guessing is default data showing through.
The date, Compare By, and Type fields get cut off after a selection is made.
The chart labels/notations are cropped in both normal and mobile views, no matter which option is selected.
Dark mode isn't applied to the chart.

<img width="1897" height="752" alt="Image" src="https://github.com/user-attachments/assets/44699e0c-9f79-4e2e-92ef-9da49643e1f1" />

Namitha7070 committed:
fix: resolve merge conflicts with development in LBDashboard

Namitha7070 committed:
Fix LBDashboard: chart labels, dark mode, metric values, filter layout, pie chart responsiveness

Namitha7070 committed:
chore: restore yarn.lock from development (no dependency changes)

HemanthNidamanuru reviewed (changes requested):
Hi Namitha,

Thank you for working on improvements, some more issues found during testing:

1. Calendar navigation arrow is not visible in light mode.
2. Two dropdown arrows are appearing in dark mode.
3. Hover values are not visible in dark mode.
4. Graph values do not change when selecting different metrics. Only the Compare By toggle (Villages/Properties) has any effect.

<img width="1903" height="926" alt="Image" src="https://github.com/user-attachments/assets/75fae2ed-7e3e-4a2a-b86e-52186cd0ecee" />

<img width="1902" height="1037" alt="Image" src="https://github.com/user-attachments/assets/592eb98e-a943-4121-9760-438ec1f47c83" />

<img width="1897" height="1041" alt="Image" src="https://github.com/user-attachments/assets/d78ee913-2e04-4d4a-af4b-3a4d0318bcf7" />

<img width="1904" height="1032" alt="Image" src="https://github.com/user-attachments/assets/8c909667-46af-4ea4-8cca-7f88c765b63b" />

Namitha7070 committed:
fix: calendar arrows, dark mode dropdowns, tooltip visibility, and filter-driven chart updates

Namitha7070 committed:
fix: resolve merge conflict - keep both ComparePieChart and RatingDistribution imports

Namitha7070 committed:
fix: resolve SonarQube quality gate issues in ComparePieChart

- Remove unused Legend import from recharts
- Remove unused CustomLegend component and its PropTypes
- Fix zero fraction numbers (1.0→1, 2.0→2, etc.) in METRIC_ITEM_MULTIPLIERS

Namitha7070 committed:
fix: add missing PropTypes for OuterPercentLabel component

- Add PropTypes validation for cx, cy, midAngle, outerRadius, percent, name, and fill props
- Resolves 14 SonarQube consistency/type-dependent issues

Namitha7070 committed:
fix: resolve 7 SonarQube issues - optional chain, zero fraction, exception handling, nested ternary, semantic HTML, cognitive complexity

Namitha7070 committed:
fix: use renderChartContent function to eliminate nested ternary and reduce cognitive complexity

HemanthNidamanuru reviewed (approved):
Approved

<img width="1896" height="1035" alt="Image" src="https://github.com/user-attachments/assets/d000384f-c849-47f8-917c-7400a6448020" />

<img width="1899" height="1037" alt="Image" src="https://github.com/user-attachments/assets/086da1fe-b697-4ea7-b192-e428afd74d29" />

<img width="1896" height="1037" alt="Image" src="https://github.com/user-attachments/assets/c4b883c6-5775-4fa1-b694-5095541e9002" />
